engine/src/configuration/applier/contact.cc

```diff
@@ -267,13 +313,207 @@ void applier::contact::modify_object(configuration::contact const& obj) {
+               ? notifier::warning
+               : notifier::none) |
+          (new_object.service_notification_options() & action_svc_critical
+               ? notifier::critical
+               : notifier::none) |
+          (new_object.service_notification_options() & action_svc_ok
+               ? notifier::ok
+               : notifier::none) |
+          (new_object.service_notification_options() & action_svc_flapping
+               ? (notifier::flappingstart | notifier::flappingstop |
+                  notifier::flappingdisabled)
+               : notifier::none) |
+          (new_object.service_notification_options() & action_svc_downtime
+               ? notifier::downtime
+               : notifier::none));
+  c->set_notify_on(
+      notifier::host_notification,
+      (new_object.host_notification_options() & action_hst_down
+           ? notifier::down
+           : notifier::none) |
+          (new_object.host_notification_options() & action_hst_unreachable
+               ? notifier::unreachable
+               : notifier::none) |
+          (new_object.host_notification_options() & action_hst_up
+               ? notifier::up
+               : notifier::none) |
+          (new_object.host_notification_options() & action_hst_flapping
+               ? (notifier::flappingstart | notifier::flappingstop |
+                  notifier::flappingdisabled)
+               : notifier::none) |
+          (new_object.host_notification_options() & action_hst_downtime
+               ? notifier::downtime
+               : notifier::none));
+  if (c->get_host_notification_period() !=
+      new_object.host_notification_period())
+    c->set_host_notification_period(new_object.host_notification_period());
+  if (c->get_service_notification_period() !=
+      new_object.service_notification_period())
+    c->set_service_notification_period(
+        new_object.service_notification_period());
+  if (c->get_host_notifications_enabled() !=
+      new_object.host_notifications_enabled())
+    c->set_host_notifications_enabled(new_object.host_notifications_enabled());
+  if (c->get_service_notifications_enabled() !=
+      new_object.service_notifications_enabled())
+    c->set_service_notifications_enabled(
+        new_object.service_notifications_enabled());
+  if (c->get_can_submit_commands() != new_object.can_submit_commands())
+    c->set_can_submit_commands(new_object.can_submit_commands());
+  if (c->get_retain_status_information() !=
+      new_object.retain_status_information())
+    c->set_retain_status_information(new_object.retain_status_information());
+  if (c->get_retain_nonstatus_information() !=
+      new_object.retain_nonstatus_information())
+    c->set_retain_nonstatus_information(
+        new_object.retain_nonstatus_information());
+  c->set_timezone(new_object.timezone());
+
+  // Host notification commands.
+  if (!MessageDifferencer::Equals(new_object.host_notification_commands(),
```

Comment:
to_modify is yet a copy of new_object

Reply:
No, it isn't, or I made an error previously.
new_object is the new version of the configuration whereas to_modify is the old version of the configuration.
So the goal here is to compare them and only copy one into the other if there are differences.